### PR TITLE
[tree-widget] refactor

### PR DESCRIPTION
**Pull request by aruniverse**
- Remove deprecated widget-control
- consolidate types
- remove containment trees as theres a replacement in breakdown-trees

### CODE REVIEW

File: packages/itwin/tree-widget/src/components/trees/SpatialTree.tsx [hunk 1]
```diff
@@ -2,27 +2,18 @@
...
+  props: SpatialTreeProps) => {
+  const iModel = useActiveIModelConnection();
 
-  return (
-    <div ref={ref} style={{ width: "100%", height: "100%" }}>
-      {width && height && (
-        <SpatialContainmentTree {...props} width={width} height={height} />
-      )}
-    </div>
+  return !iModel ? null : (
```
Comment (jason-crow) [inline packages/itwin/tree-widget/src/components/trees/SpatialTree.tsx:14]: seems more straightforward to do:

iModel && (<AutoSizer> ... </AutoSizer>)

Comment (aruniverse) [reply packages/itwin/tree-widget/src/components/trees/SpatialTree.tsx:14]: Agreed, but that causes issues with the SelectableContentDefintion.
image
image

Comment (aruniverse) [reply packages/itwin/tree-widget/src/components/trees/SpatialTree.tsx:14]: Eh actually nvm, i guess I can just wrap the entire thing in a react fragment so its a valid ReactNode; Ill update

File: packages/itwin/tree-widget/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import type {
+  CategoryTreeProps,
+  ModelsTreeProps,
+  SpatialContainmentTreeProps,
+} from "@itwin/appui-react";
+import type { SelectableContentDefinition } from "@itwin/components-react";
+import type { IModelConnection } from "@itwin/core-frontend";
+
+export interface ClassificationsTreeComponentProps {
```
Comment (jason-crow) [inline packages/itwin/tree-widget/src/types.ts:9]: arent you removing the ClassificationsTree?

File: packages/itwin/tree-widget/src/components/trees/CategoriesTree.tsx [hunk 3]
```diff
@@ -76,40 +65,41 @@ export function CategoriesTreeComponent(props: CategoriesTreeComponentProps) {
...
       disabled,
       true,
       true
     );
-  }, [props.iModel, filteredProvider]);
+  }, [iModel, viewport, filteredProvider]);
 
-  return (
+  return (!(iModel && viewport) ? null : (
     <>
```
Comment (jason-crow) [inline packages/itwin/tree-widget/src/components/trees/CategoriesTree.tsx:84]: <>
{ iModel && viewport && (<AutoSizer>...</AutoSizer>) }
</>

**Review by jason-crow — APPROVED** [review]

**Pull request merged**